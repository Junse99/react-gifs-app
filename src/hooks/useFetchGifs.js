import { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs'


export const useFetchGifs = ( category ) => {

  const [state, setState] = useState({
    data: [],
    loading: true
  })

  //Para que el getGifts se ejecute una sola vez al inicio, es necesario el [] al final que son las dependencias
  useEffect(() => {

    getGifs( category )
      .then(imgs => setState({
        data: imgs,
        loading: false
      }))

  }, [ category ])//Si la category cambia, se ejecute otra vez el useEffect */

  return state

}