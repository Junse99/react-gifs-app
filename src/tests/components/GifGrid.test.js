import React from 'react';
import { shallow } from "enzyme"
import GiftGrid from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')//Esto finge que llama a ese archivo para traer los datos

describe('Pruebas de <GifGrid/>', () => {

  const category = 'Terror'

  test('debe de mostrarse correctamente', () => {
    //Con esto finge que llama el fetchGifs, pero devuelve este objeto literal que se especifica aqui
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GiftGrid category={ category }/>)

    expect( wrapper ).toMatchSnapshot()
  })

  test('debe de mostrar items cuando se cargan imagenes con useFetchGifs()', () => {

    const gifs = [{
      id: 'ABC',
      title: 'Prueba jeje',
      url: 'https://cualquiercosa.com'
    }]
    
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })


    const wrapper = shallow(<GiftGrid category={ category }/>)
    expect( wrapper ).toMatchSnapshot()
    //Verificar que el parrafo de caragando no exista
    expect( wrapper.find('p').exists()).toBe( false )
    //Veriicar si un GifGridItem existe de acuerdo a los gifs del arreglo
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )

  })

  
  
  
})
