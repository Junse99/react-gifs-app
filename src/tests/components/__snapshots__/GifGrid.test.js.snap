// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid/> debe de mostrar items cuando se cargan imagenes con useFetchGifs() 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Terror
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Prueba jeje"
      url="https://cualquiercosa.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de <GifGrid/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Terror
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
